docs/api-conventions.md

```diff
@@ -0,0 +1,43 @@
+# UCP API conventions
+A collection of conventions that components of the UnderCloud Platform (UCP) utilize for their REST APIs
+---
+## Resource path naming
+* Resource paths nodes follow an all lower case naming scheme, and pluralize the resource names. Nodes that refer to keys, ids or names that are externally controlled, the external naming will be honored.
+* The version of the API resource path will be prefixed before the first node of the path for that resource using v#.# format.
+* By default, the API will be namespaced by /api before the version.  For the purposes of documentation, this will not be specified in each of the resource paths below.  In more complex APIs, it makes sense to allow the /api node to be more specific to point to a particular service.
+```
+/api/v1.0/sampleresources/ExTeRnAlNAME-1234
+      ^         ^       ^       ^
+      |         |       |      defer to external naming
+      |         |      plural
+      |        lower case
+     version here
+```
+---
+## Error responses
+Error responses (HTTP response body accompanying 4xx series responses) are a specialization of the [Kubernetes standard for error representation](https://github.com/kubernetes/community/blob/master/contributors/devel/api-conventions.md#response-status-kind). 
```

Comment:
Should this not also apply to 5XX status codes. Nothing better than getting a blank-bodied 500 back.